src/filebrowser/browser.ts

```diff
@@ -106,10 +106,11 @@ class FileBrowserWidget extends Widget {
 
     model.refreshed.connect(this._handleRefresh, this);
     this._crumbs = new BreadCrumbs({ model });
+    this._listing = new DirListing({ manager, model, opener, renderer });
+    let listings = this._listing;
     this._buttons = new FileButtons({
-      commands, keymap, manager, model, opener
+      commands, keymap, manager, model, opener, listings
     });
-    this._listing = new DirListing({ manager, model, opener, renderer });
 
     model.fileChanged.connect((fbModel, args) => {
```

Comment:
This is where you will modify the logic to add calls to the needed listing methods. You will add a test for `args.type` to see if it is a new directory and then take the needed action